### PR TITLE
[Merged by Bors] - chore(Data/Polynomial/Degree/TrailingDegree): some basic api lemmas

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Polynomial/Degree/TrailingDegree.lean [hunk 1]
```diff
@@ -512,6 +522,22 @@ theorem ne_zero_of_trailingDegree_lt {n : ℕ∞} (h : trailingDegree p < n) : p
...
+lemma Monic.eq_X_pow_of_natTrailingDegree_eq_natDegree
+    (h₁ : p.Monic) (h₂ : p.natTrailingDegree = p.natDegree) :
+    p = X ^ p.natDegree := by
+  ext n
+  rw [coeff_X_pow]
+  obtain hn|rfl|hn := lt_trichotomy n p.natDegree
+  · rw [if_neg hn.ne, coeff_eq_zero_of_lt_natTrailingDegree]
+    rwa [h₂]
+  · rw [if_pos rfl]
+    exact h₁.leadingCoeff
```
Comment (acmepjz) [inline Mathlib/Data/Polynomial/Degree/TrailingDegree.lean:538]:
```suggestion
  obtain hn | rfl | hn := lt_trichotomy n p.natDegree
  · rw [if_neg hn.ne, coeff_eq_zero_of_lt_natTrailingDegree (h₂ ▸ hn)]
  · simpa only [if_pos rfl] using h₁.leadingCoeff
```

Personally, I think add some spaces between `|` and identifiers are better.

File: Mathlib/Data/Polynomial/Degree/TrailingDegree.lean [hunk 2]
```diff
@@ -450,6 +450,16 @@ theorem natTrailingDegree_X : (X : R[X]).natTrailingDegree = 1 :=
 set_option linter.uppercaseLean3 false in
 #align polynomial.nat_trailing_degree_X Polynomial.natTrailingDegree_X
 
+@[simp]
+lemma trailingDegree_X_pow (n : ℕ) :
+    (X (R := R) ^ n).trailingDegree = n := by
```
Comment (j-loreaux) [inline Mathlib/Data/Polynomial/Degree/TrailingDegree.lean:455]: Personally, I find `(X ^ n : R[X]) more readable, but this is optional.
```suggestion
    (X ^ n : R[X]).trailingDegree = n := by
```

File: Mathlib/Data/Polynomial/Degree/TrailingDegree.lean [hunk 3]
```diff
@@ -450,6 +450,16 @@ theorem natTrailingDegree_X : (X : R[X]).natTrailingDegree = 1 :=
...
+  rw [X_pow_eq_monomial, trailingDegree_monomial one_ne_zero]
+
+@[simp]
+lemma natTrailingDegree_X_pow (n : ℕ) :
+    (X (R := R) ^ n).natTrailingDegree = n := by
```
Comment (j-loreaux) [inline Mathlib/Data/Polynomial/Degree/TrailingDegree.lean:460]: same
```suggestion
    (X ^ n : R[X]).natTrailingDegree = n := by
```

**Comment by j-loreaux** [comment]
bors d+

**Review by j-loreaux — COMMENTED** [review]
oops, I meant to leave these before delegation.

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**